### PR TITLE
add `--active_class_weight` argument for allowing reweighting to account for class imbalance

### PR DESCRIPTION
**Pull request by willy-b**
Optionally, allow re-weighting of loss to account for class imbalance (this dataset has an active rate of less than 1 in 500 molecules). 
Default of 1.0 has no effect,
setting to 1/active_molecule_prevalence would encourage recall of active molecules same as inactive at cost of precision. When judging by ROCAUC and/or AP using the raw y prediction scores to rank the results, this is not necessary but may of interest especially if using output as a classifier on a single example or online rather than ranking a list.

I am not using this yet but figured may be useful for comparison / exploration.

### CODE REVIEW

**Comment by willy-b** [comment]
Different values of this hyperparameter are under testing against train/valid sets now as part of hyperparameter search, I will update the default value once the results are in. It looks like based on the first 10 runs using train/valid that for example a value of 2 (but not some much higher values, like e.g. 15) may have superior performance compared to the default of 1. The performance of `--active_class_weight 2` is now being more deeply checked with random subsets of training data held out as an additional generalization check with statistical comparison against the baseline, while basic train/valid runs are still being done for various other values of this new hyperparameter.
